src/actions/MastodonModelAction.C

```diff
@@ -41,9 +41,13 @@ validParams<MastodonModelAction>()
                              "acceleration AuxVariables and AuxKernels, inertia kernels and the "
                              "DynamicTensorMechanicsAction based on the dimension of the mesh and "
                              "if static or dynamic analysis is being performed.");
+  params.addParam<std::vector<SubdomainName>>("block",
+                                              "A list of ids of the blocks (subdomains) "
+                                              "that these model parameters apply to.");
   params.addParam<bool>("dynamic_analysis", true, "false, if static analysis is to be performed.");
   params.addParam<Real>("beta", 0.25, "beta parameter for Newmark time integration.");
   params.addParam<Real>("gamma", 0.5, "gamma parameter for Newmark time integration.");
+  params.addParam<unsigned int>("dim", 3, "Dimension of the mesh.");
```

Comment:
You should make this a MooseEnum, then you won't need to do the error check below.

```cpp
MooseEnum dim("1 2 3", "3");
params.addParam<MooseEnum>("dim", dim, ...);
```